planner/core/plan_cost_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/pingcap/tidb/testkit"
+)
+
+func checkCost(t *testing.T, tk *testkit.TestKit, q, info string) {
+	//| id | estRows | estCost   | task | access object | operator info |
+	tk.MustExec(`set @@tidb_enable_new_cost_interface=0`)
+	rs := tk.MustQuery("explain format=verbose " + q).Rows()
+	oldRoot := fmt.Sprintf("%v", rs[0])
+	oldPlan := ""
+	for _, r := range rs {
+		oldPlan = oldPlan + fmt.Sprintf("%v\n", r)
+	}
+	tk.MustExec(`set @@tidb_enable_new_cost_interface=1`)
+	rs = tk.MustQuery("explain format=verbose " + q).Rows()
+	newRoot := fmt.Sprintf("%v", rs[0])
+	newPlan := ""
+	for _, r := range rs {
+		newPlan = newPlan + fmt.Sprintf("%v\n", r)
+	}
+	if oldRoot != newRoot {
+		t.Fatalf("run %v failed, info: %v, expected \n%v\n, but got \n%v\n", q, info, oldPlan, newPlan)
+	}
+}
+
+func TestNewCostInterfaceTiKV(t *testing.T) {
```

Comment:
It's better not to have such heavy test cases in the unit tests.

Reply:
Updated. Now, this test won't insert any data into the table and just uses `pseudo-stats` to test.